test/app-luatest/gh_6200_lua_cjson_escapes_slash_test.lua

```diff
@@ -0,0 +1,104 @@
+local compat = require('compat')
+local misc = require('test.luatest_helpers.misc')
+local jit = require('jit')
+local json = require('json')
+local popen = require('popen')
+local clock = require('clock')
+local t = require('luatest')
+local g = t.group()
+
+g.test_json_encode = function()
+    compat.json_escape_forward_slash = 'old'
+    -- Test that '/' is escaped with 'old' setting.
+    t.assert_equals(json.encode({url = 'https://srv:7777'}), '{"url":"https:\\/\\/srv:7777"}')
+    t.assert_equals(json.encode('/home/user/tarantool'), [["\/home\/user\/tarantool"]])
+    -- Test that other escape symbols are not affected by the setting.
+    t.assert_equals(json.encode('\t'), [["\t"]])
+    t.assert_equals(json.encode('\\'), [["\\"]])
+
+    compat.json_escape_forward_slash = 'new'
+    -- Test that '/' is not escaped with 'new' setting.
+    t.assert_equals(json.encode({url = 'https://srv:7777'}), [[{"url":"https://srv:7777"}]])
+    t.assert_equals(json.encode('/home/user/tarantool'), [["/home/user/tarantool"]])
+    -- Test that other escape symbols are not affected by the setting.
+    t.assert_equals(json.encode('\t'), [["\t"]])
+    t.assert_equals(json.encode('\\'), [["\\"]])
+
+    -- Restore options defaults.
+    compat.json_escape_forward_slash = 'default'
+end
+
+g.test_json_new_encode = function()
+    compat.json_escape_forward_slash = 'old'
+    -- Test that '/' is escaped with 'old' setting.
+    t.assert_equals(json.encode('/'), [["\/"]])
+
+    -- Create new serializer and check that it has correct defaults and doesn't change behavior.
+    local json_old = json.new()
+    t.assert_equals(json_old.encode('/'), [["\/"]])
+    compat.json_escape_forward_slash = 'new'
+    t.assert_equals(json_old.encode('/'), [["\/"]])
+    t.assert_equals(json.encode('/'), [["/"]])
+
+    -- Create new serializer and check that it has correct defaults.
+    local json_new = json.new()
+    t.assert_equals(json_new.encode('/'), [["/"]])
+
+    -- Restore options defaults.
+    compat.json_escape_forward_slash = 'default'
+end
+
+-- log.info in json format mode uses internal encoder that takes msgpuck
+-- char2escape table, while mp_char2escape is changed by compat.
+g.test_mp_encode = function()
+    misc.skip_if_not_debug()
+    t.skip_if(jit.os == 'OSX', 'popen is broken on OSX.')
+    -- Start external tarantool session.
+    local TARANTOOL_PATH = arg[-1]
+    local cmd = 'ERRINJ_STDIN_ISATTY=1 ' .. TARANTOOL_PATH .. ' -i 2>&1'
```

Comment:
This error injection was removed in 16d6e9d2e272e3c7fba807ef64a8aac791e2c427.

BTW, you can find `interactive_tarantool.lua` from PR #7170 convenient.